### PR TITLE
ipt: implement CATCHALL_BREAKPOINT trap type

### PR DESCRIPTION
**Pull request by icedevml**
Because for the IPT we need to inject a special PTWRITE packet whenever a DRAKVUF breakpoint is hit, the IPT plugin needs to have ability to subscribe to all breakpoints. It doesn't need any special insight about what other plugins do, but it needs such kind of event to properly synchronize the IPT logs.

This might look weird but I hope reasonable.

### CODE REVIEW

**Comment by tklengyel** [comment]
Makes sense

**Comment by tklengyel** [comment]
For such handlers in LibVMI we use the term "generic" instead of global. I think we should stick to that here too.

**Comment by tklengyel** [comment]
Or I would also be happy with something like "ALL_BREAKPOINTS" or "CATCHALL_BREAKPOINT"

**Comment by icedevml** [comment]
> Or I would also be happy with something like "ALL_BREAKPOINTS" or "CATCHALL_BREAKPOINT"

I've modified the name to `CATCHALL_BREAKPOINT`

File: src/libdrakvuf/vmi.c [hunk 1]
```diff
@@ -619,12 +619,16 @@ event_response_t int3_cb(vmi_instance_t vmi, vmi_event_t* event)
     trap_info.attached_proc_data.tid       = attached_proc_data.tid;
 
     drakvuf->in_callback = 1;
-    GSList* loop = s->traps;
-    while (loop)
+    GSList* lists[2] = {drakvuf->catchall_breakpoint, s->traps};
```
Comment (icedevml) [inline src/libdrakvuf/vmi.c:622]: Skip if `s->traps` is empty, no need to fire `CATCHALL_BREAKPOINT` in this case

**Comment by icedevml** [comment]
this one is ready

**Comment by icedevml** [comment]
@drakvuf-jenkins Retest this please.

**Pull request merged**